migrate/3.0-restore.sh

```diff
@@ -79,24 +95,40 @@ function circleci_database_import() {
 
     preflight_checks
 
-    echo "...scaling application deployments to 0..."
-    scale_deployments 0
-
-    # wait one minute for pods to scale down
-    sleep 60
+    if [ ! "$SKIP_MONGO $SKIP_POSTGRES $SKIP_VAULT" = "true true true" ];
+    then
+        echo "...scaling application deployments to 0..."
+        scale_deployments 0
+        
+        # wait one minute for pods to scale down
+        sleep 60
+    fi
 
-    if [ ! "$SKIP_DATABASE_IMPORT" = "true" ];
+    if [ ! "$SKIP_MONGO" = "true" ];
     then
-        import_postgres
+        MONGO_BU="${BACKUP_DIR}/circleci-mongo-export"
         import_mongo
     fi
     
-    import_vault
+    if [ ! "$SKIP_POSTGRES" = "true" ];
+    then
+        PG_BU="${BACKUP_DIR}/circleci-pg-export"
+        import_postgres
+    fi
+
+    if [ ! "$SKIP_VAULT" = "true" ];
+    then
+        VAULT_BU="${BACKUP_DIR}/circleci-vault"
+        import_vault
+    fi
 
-    reinject_bottoken
+    if [ ! "$SKIP_MONGO $SKIP_POSTGRES $SKIP_VAULT" = "true true true" ];
+    then
+        reinject_bottoken
```

Comment:
Wouldn't it be better to move the `reinject_bottoken` step into the block that imports the Mongo data?

Reply:
I'm honestly not that familiar with inject bottoken but I'll take your word on it